src/error.rs

```diff
@@ -24,6 +84,23 @@ impl Error {
     ///
     /// The `'static` bounds allows using `downcast_ref` to check the
     /// details of the error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # extern crate reqwest;
+    /// # fn main() {
+    /// let invalid_request = "dogs://puppy";
+    /// let response = reqwest::get(invalid_request);
+    /// match response {
+    ///   Err(e) => match e.get_ref() {
+    ///     Some(internal_error) => println!("Problem with request: {}", internal_error),
```

Comment:
I'd guess that @dtolnay meant to show a typical example of why someone would actually use this method. You can already print the `reqwest::Error`, so why would a user want to call `get_ref`?

A possible reason could be to try to handle certain cases, like casting to a `UrlError` and inspecting exactly what was wrong with the provide URL.